src/main/java/dev/openfeature/sdk/AbstractStructure.java

```diff
@@ -18,7 +19,15 @@ public boolean isEmpty() {
     }
 
     AbstractStructure(Map<String, Value> attributes) {
-        this.attributes = new HashMap<>(attributes);
+        this.attributes = attributes;
+    }
+
+    /**
+     * Returns an unmodifiable representation of the internal attribute map.
+     * @return immutable map
+     */
+    public Map<String, Value> asUnmodifiableMap() {
+        return Collections.unmodifiableMap(attributes);
```

Comment:
Is it worth to check if this is an unmodifiable nap, before wrapping it?

Reply:
According to SO, `Collections.unmodifiableMap` already does this.